src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -727,6 +727,8 @@ protected DynamoModel(IStartConfiguration config)
             LibraryServices.MessageLogged += LogMessage;
             LibraryServices.LibraryLoaded += LibraryLoaded;
 
+            EngineController.VMLibrariesReset += ReloadDummyNodes;
```

Comment:
I think this should be done after the call to `ResetEngineInternal` as a new `EngineController` could be created there, in which case this event will not be subscribed by the new controller.

Reply:
Moved it.